src/Page.js

```diff
@@ -449,6 +450,11 @@ Page.prototype.insertFooter = function (pageData) {
   return `${pageData}\n${nunjucks.renderString(footerContent, userDefinedVariables)}`;
 };
 
+
+Page.prototype.insertSpinner = function (pageData) {
+  return `${pageData}\n${SPINNER_HTML}`;
+};
```

Comment:
The addition of the spinner need not be done through code, because the HTML content is always going to be the same.

Please add the spinner directly inside `page.ejs` instead.